packages/dart/npt_flutter/lib/features/onboarding/util/profile_progress_listener.dart

```diff
@@ -0,0 +1,20 @@
+import 'dart:developer';
+
+import 'package:at_client_mobile/at_client_mobile.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:npt_flutter/app.dart';
+import 'package:npt_flutter/features/profile_list/bloc/profile_list_bloc.dart';
+
+class ProfileProgressListener extends SyncProgressListener {
+  @override
+  void onSyncProgressEvent(SyncProgress syncProgress) {
+    final profileListBlock = App.navState.currentContext!.read<ProfileListBloc>();
+
+    final profileListState = profileListBlock.state as ProfileListLoaded;
```

Comment:
There's no checks in place to confirm that profileListBlock.state is actually the ProfileListLoaded type, what if it's ProfileListInitial, then it throws an uncaught exception.


The if below should be something like 

```dart
... SyncStatus.success && (profileListBlock.state is! ProfileListLoaded || (profileListBlock.state as ProfileListLoaded).profiles.isEmpty)
```

Reply:
Then change `is!` to `is` and the following `||` to `&&`. Either way, you need to check the type before casting.